cli/test_pack_python3.bats

```diff
@@ -42,20 +43,15 @@ skip_tests_if_python3_is_not_available_or_if_already_running_under_python3() {
 @test "packs.setup_virtualenv without python3 flags works and defaults to Python 2" {
 	skip_tests_if_python3_is_not_available_or_if_already_running_under_python3
 
-	SETUP_VENV_RESULTS=$(st2 run packs.setup_virtualenv packs=examples -j)
+	SETUP_VENV_RESULTS=$(run_command_and_log_output st2 run packs.setup_virtualenv packs=examples -j)
 	run eval "echo '$SETUP_VENV_RESULTS' | jq -r '.result.result'"
 	assert_success
-
 	assert_output "Successfully set up virtualenv for the following packs: examples"
```

Comment:
For example, in this scenario avoiding another `run` and mutating output will help to preserve original command results. We do assert against the partial output, `jq` here is overcomplication:
```diff
-	run eval "echo '$SETUP_VENV_RESULTS' | jq -r '.result.result'"
-	assert_success
-	assert_output "Successfully set up virtualenv for the following packs: examples"
+	assert_output --partial "Successfully set up virtualenv for the following packs: examples"
```

Reply:
That change is fine, but in addition to checking for a specific string, we should also check that the output is valid JSON.

One of the commands in one of the BATS tests returned a JSON array concatenated to a JSON dict:

```json
[
  "example_element_1",
  "example_element_2"
]
{
  "example_key_1": "example_value_1",
  "example_key_2": "example_value_2"
}
```

Both of those objects individually are valid JSON, but simply concatenating them together produces invalid JSON.

So we do still need some way to validate that the produced output is valid JSON. This might be good to do in a custom `assert_` function, and could even simply wrap `jq`.

But however we check for JSON validity, this simplification is a good one.